reconcile/ocm_addons.py

```diff
@@ -0,0 +1,88 @@
+import sys
+import logging
+
+import reconcile.queries as queries
+
+from reconcile.status import ExitCodes
+from utils.ocm import OCMMap
+
+QONTRACT_INTEGRATION = 'ocm-addons'
+
+
+def fetch_current_state(clusters):
+    settings = queries.get_app_interface_settings()
+    ocm_map = OCMMap(clusters=clusters, integration=QONTRACT_INTEGRATION,
+                     settings=settings)
+
+    current_state = []
+    for cluster in clusters:
+        cluster_name = cluster['name']
+        ocm = ocm_map.get(cluster_name)
+        addons = ocm.get_cluster_addons(cluster_name)
+        if addons:
+            for addon in addons:
+                addon['cluster'] = cluster_name
+                current_state.append(addon)
+
+    return ocm_map, current_state
+
+
+def fetch_desired_state(clusters):
+    desired_state = []
+    for cluster in clusters:
+        cluster_name = cluster['name']
+        addons = cluster['addons']
+        for addon in addons:
+            addon['id'] = addon.pop('name')
+            addon['cluster'] = cluster_name
+            desired_state.append(addon)
+
+    return desired_state
+
+
+def calculate_diff(current_state, desired_state):
+    diffs = []
+    for d in desired_state:
+        c = [c for c in current_state
+             if d.items() <= c.items()]
+        if not c:
+            d['action'] = 'install'
+            diffs.append(d)
+
+    return diffs
+
+
+def act(dry_run, diffs, ocm_map):
+    err = False
+    for diff in diffs:
+        action = diff.pop('action')
+        cluster = diff.pop('cluster')
```

Comment:
I'd rather be explicit here and create a new dict to pass to `install_addon` than modifying the given structure while iterating over it... It makes things harder to follow, to debug and most likely they're more prone to errors.

Reply:
we add the `cluster` and the `action` in the functions right above. i kind of like this approach, because it allows us to add fields in the app-interface schema without changing the integration's code.